core/src/main/java/io/fabric8/maven/core/service/BuildService.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.service;
+
+import java.io.File;
+
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.maven.core.config.BuildRecreateMode;
+import io.fabric8.maven.core.config.OpenShiftBuildStrategy;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.docker.util.MojoParameters;
+import io.fabric8.maven.docker.util.Task;
+
+import org.apache.maven.plugins.annotations.Parameter;
+
+/**
+ * @author nicola
+ * @since 17/02/2017
+ */
+public interface BuildService {
+
+    /**
+     * Builds the given image using the specified configuration.
+     *
+     * @param config the build configuration
+     * @param imageConfig the image to build
+     */
+    void build(BuildServiceConfig config, ImageConfiguration imageConfig) throws Fabric8ServiceException;
+
+
+    /**
+     * Class to hold configuration parameters for the building service.
+     */
+    class BuildServiceConfig {
+
+        private io.fabric8.maven.docker.service.BuildService.BuildContext dockerBuildContext;
+
+        private MojoParameters dockerMojoParameters;
```

Comment:
Agree. In order to remove it we should first simplify the build service in `d-m-p`, then come back here.